src/transport/UnauthenticatedSessionTable.h

```diff
@@ -59,13 +59,27 @@ class UnauthenticatedSession : public ReferenceCounted<UnauthenticatedSession, U
 
     const PeerAddress & GetPeerAddress() const { return mPeerAddress; }
 
+    void GetMRPIntervals(uint32_t & idleInterval, uint32_t & activeInterval)
+    {
+        idleInterval   = mMRPIdleInterval;
+        activeInterval = mMRPActiveInterval;
+    }
+
+    void SetMRPIntervals(uint32_t idleInterval, uint32_t activeInterval)
+    {
+        mMRPIdleInterval   = idleInterval;
+        mMRPActiveInterval = activeInterval;
+    }
+
     MessageCounter & GetLocalMessageCounter() { return mLocalMessageCounter; }
     PeerMessageCounter & GetPeerMessageCounter() { return mPeerMessageCounter; }
 
 private:
     uint64_t mLastActivityTimeMs = 0;
 
     const PeerAddress mPeerAddress;
+    uint32_t mMRPIdleInterval   = 0;
```

Comment:
Maybe initialize both sessions with `CHIP_CONFIG_MRP_DEFAULT_INITIAL_RETRY_INTERVAL` and `CHIP_CONFIG_MRP_DEFAULT_ACTIVE_RETRY_INTERVAL`?

Reply:
I would like to, but those configs are part of the messaging module that cannot be dependency of a transport module (transport already is a dependency of messaging). I thought that setting it to 0 will not be a bad idea, as anyway we set it every time when new session is created.

Reply:
Ok, I'm kinda afraid that someone may use the session in a different context one day and forget about setting these values. But maybe it's OK as he will notice it very quickly ;).